std/file.d

```diff
@@ -570,6 +628,23 @@ void rename(RF, RT)(RF from, RT to)
     renameImpl(f, t, fromz, toz);
 }
 
+/// ditto
+void rename(RF, RT)(auto ref RF from, auto ref RT to)
+    if (isStringLike!RF || isStringLike!RT)
+{
+    import std.meta : staticMap;
+    alias selectString(R) = Select!(isStringLike!R, StringTypeOf!R, R);
+    alias Types = staticMap!(selectString, RF, RT);
+    rename!Types(from, to);
+}
```

Comment:
Interesting solution. Not sure if it's important, but error messages can get confusing due to the delayed constraints:
```
import std.file;

void foo()
{
	DirEntry de;
	rename(de, 1);
}
```
generates an error that looks like a bug in phobos:
```
c:\s\d\rainers\phobos\std\traits.d(4913): Error: static assert  "int is not a string type"
c:\s\d\rainers\phobos\std\file.d(636):        instantiated from here: StringTypeOf!int
c:\s\d\rainers\phobos\std\meta.d(546):        instantiated from here: F!int
c:\s\d\rainers\phobos\std\meta.d(553):        instantiated from here: staticMap!(selectString, int)
c:\s\d\rainers\phobos\std\file.d(637):        instantiated from here: staticMap!(selectString, DirEntry, int)
test.d(7):        instantiated from here: rename!(DirEntry, int)
```